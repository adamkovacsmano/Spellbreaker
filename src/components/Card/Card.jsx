import React, { Component } from "react";
import styles from "./Card.module.scss";

class Card extends Component {
  state = {};
  render() {
    return (
      <div className={styles.cardContainer} style={{cursor: "pointer"}}>
        <h1>{this.props.playername} {this.props.name}</h1>
        <img src={this.props.img} alt="character-pic"></img>
        <section>
          <p>{this.props.description}</p>
        </section>
        <section className={styles.attributes}>
          <table>
            <tr>
              <th>Combat expertise</th>
              <td>{this.props.combatexp}</td>
            </tr>
            <tr>
              <th>Willpower</th>
              <td>{this.props.willpower}</td>
            </tr>
            <tr>
              <th>Luck</th>
              <td>{this.props.luck}</td>
            </tr>
          </table>
        </section>
        <section>
          <tr className={styles.hp}>
            <tr>HP:</tr>
            <td>{this.props.hp}</td>
          </tr>
        </section>
      </div>
    );
  }
}

export default Card;
