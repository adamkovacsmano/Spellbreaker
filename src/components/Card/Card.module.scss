.cardContainer {
  width: 350px;
  height: 100%;
  border: 4px solid black;
  border-radius: 10px;
  box-sizing: border-box;
  padding: 10px;
  text-align: center;
  background-image: url(../../Images/cardbackground.jpg);
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;

  h1 {
    font-size: 1.5em;
    color: rgb(255, 60, 60);
  }

  p {
    border: 1px solid white;
    padding: 5px;
    color: white;
  }

  img {
    width: 100%;
    border: 1px solid black;
  }

  table {
    width: 100%;
    padding: 5px;
  }
}

.attributes {
  vertical-align: middle;

  th {
    color: white;
    background-color: black;
    border-bottom-right-radius: 20px;
    border: 1px solid white;
    text-align: left;
    padding: 5px;
  }

  td {
    color: white;
    background-color: black;
    border-top-left-radius: 20px;
    border: 1px solid white;
    text-align: right;
    padding: 5px;
  }
}

.hp {
  display: flex;
  justify-content: flex-end;
  background-color: black;
  color: white;
  padding: 5px 10px;
  border: 1px solid white;

  tr {
    padding-right: 5px;
  }

  td {
    color: rgb(255, 60, 60);
  }
}
