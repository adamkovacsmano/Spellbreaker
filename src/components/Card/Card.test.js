import React from "react";
import { shallow } from "enzyme";
import Card from "./Card";

describe("CreatePost tests", () => {
  let card;
  beforeEach(() => {
    card = shallow(<Card />);
  });

  test("should match the snapshot", () => {
    expect(card).toMatchSnapshot();
  });
});
