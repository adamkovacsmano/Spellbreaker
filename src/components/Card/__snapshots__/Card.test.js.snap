// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePost tests should match the snapshot 1`] = `
<div
  className="cardContainer "
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
    }
  }
>
  <h1>
     
  </h1>
  <img
    alt="character-pic"
  />
  <section>
    <p />
  </section>
  <section
    className="attributes"
  >
    <table>
      <tr>
        <th>
          Combat expertise
        </th>
        <td />
      </tr>
      <tr>
        <th>
          Willpower
        </th>
        <td />
      </tr>
      <tr>
        <th>
          Luck
        </th>
        <td />
      </tr>
    </table>
  </section>
  <section>
    <tr
      className="hp"
    >
      <tr>
        HP:
      </tr>
      <td />
    </tr>
  </section>
</div>
`;
